Drv/Ip/test/ut/TestUdp.cpp

```diff
@@ -15,31 +15,27 @@ void test_with_loop(U32 iterations, bool duplex) {
     Drv::SocketIpStatus status1 = Drv::SOCK_SUCCESS;
     Drv::SocketIpStatus status2 = Drv::SOCK_SUCCESS;
 
-    U16 port1 =  Drv::Test::get_free_port();
-    ASSERT_NE(0, port1);
+    // When not duplex, we can allow the OS to choose a port
+    U16 port1 =  (duplex) ? Drv::Test::get_free_port() : 0;
```

Comment:
Why don't we allow the OS to choose also in duplex configuration? And couldn't we do that for both ports?

That might avoid having equality between `port1` and `port2`

Reply:
In order to use "0", `udp1.configureSend` must come after `udp2.open` and `udp2.configureSend` must come after `udp2.open`. `udpX.configureSend` and `udpX.configureRecv` must come before `udpX.open`.  

Thus there is a dependency loop.  In order to ensure there is a test for "0" I worked it into the half-duplex test, but could not see an easy way to sort it out for full-duplex.